smartsim/settings/launch_settings.py

```diff
@@ -126,6 +126,18 @@ def __init__(
             """The launcher type"""
         except ValueError:
             raise ValueError(f"Invalid launcher type: {launcher}")
+
```

Comment:
I notice that there hasnt been Value Checking added to arguments/launch/ modules, but all Value checking has been added to batch modules, is launch_setting value checking also going into this PR?